modules/serverless-config-high-workloads-serving.adoc

```diff
@@ -0,0 +1,68 @@
+= Configuring Serving for high workloads
+
+Findings relevant to configuring Knative Serving for a high workload:
+
+[NOTE]
+====
+These findings have been tested with requests with a payload size of 0-32 kb. The Knative Service backends used in those tests had a startup-latency between 0-10 seconds and response times between 0-5 seconds.
+====
+
+* All data-plane components are mostly increasing CPU usage on higher requests and/or payload scenarios, so the CPU requests and limits have to be tested and potentially increased.
+* The activator component also might need more memory, when it has to buffer more or bigger request payloads, so the memory requests and limits might need to be increased as well.
+* One activator pod can handle approximately 2500 requests per second before it starts to increase latency and, at some point, leads to errors.
+* One 3scale-kourier-gateway or istio-ingressgateway pod can also handle approximately 2500 requests per second before it starts to increase latency and, at some point, leads to errors.
+* Each of the data-plane components consumes up to 1 vCPU of CPU for handling 2500 requests per second. Note that this highly depends on the payload size and the response times of the Knative Service backend.
+
+[IMPORTANT]
+====
+Fast startup and fast response-times of your Knative Service user workloads are critical for good performance of the overall system. As Knative Serving components are buffering incoming requests when the Knative Service user backend is scaling-up or request concurrency has reached its capacity. If your Knative Service user workload introduces long startup or request-latency, it will either overload the `activator` component (when the CPU + memory configuration is too low) or lead to errors for the calling clients.
+====
+
+To fine-tune your installation, use the above findings combined with your own test results to configure the `KnativeServing` custom resource:
+
+
+[source,yaml]
+----
+apiVersion: operator.knative.dev/v1beta1
+kind: KnativeServing
+metadata:
+  name: knative-serving
+  namespace: knative-serving
+spec:
+  high-availability:
+    replicas: 2 <1>
+  workloads:
+    - name: component-name <2>
+      replicas: 2 <2>
+      resources:
+        - container: container-name
+          requests:
+            cpu: <3>
+            memory: <3>
+          limits:
+            cpu: <3>
+            memory: <3>
+  podDisruptionBudgets: <4>
+    - name: name-of-pod-disruption-budget
+      minAvailable: 1
+----
+<1> Set this to at least 2, to make sure you have always at least two instances of every component running. You can also use `workloads` to override the replicas for certain components.
+<2> Use the `workloads` list to configure specific components. Use the `deployment` name of the component (like `activator`, `autoscaler`, `autoscaler-hpa`, `controller`, `webhook`, `net-kourier-controller`, `3scale-kourier-gateway`, `net-istio-controller`) and set the `replicas`.
```

Comment:
We should also add a note that would say something like:

For components that use HorizontalPodAutoscalers  (activator, webhook and 3scale-kourier-gateway) , the `replicas` field sets the _minimum_ number of replicas. Actual number of replicas will depend on the CPU load and scaling done by the HPAs.